python/python/tests/test_async_client.py

```diff
@@ -1934,18 +1934,38 @@ def clean_result(value: TResult):
     async def test_cluster_fail_routing_by_address_if_no_port_is_provided(
         self, redis_client: RedisClusterClient
     ):
-        with pytest.raises(RequestError) as e:
+        with pytest.raises(RequestError):
             await redis_client.info(route=ByAddressRoute("foo"))
 
 
 @pytest.mark.asyncio
 class TestExceptions:
     @pytest.mark.parametrize("cluster_mode", [True, False])
     @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
-    async def test_timeout_exception_with_blpop(self, redis_client: TRedisClient):
-        key = get_random_string(10)
+    async def test_timeout_exception_with_long_running_lua(
+        self, redis_client: TRedisClient, request
+    ):
+        # Create a client with minimal request timeout
+        is_cluster = isinstance(redis_client, RedisClusterClient)
+        new_client = await create_client(
+            request,
+            is_cluster,
+            addresses=redis_client.config.addresses,
+            request_timeout=1,
+        )
+        script = Script(
+            """
+            local i = 0
+            while (i < 100000)
+            do
+            redis.call('ping')
```

Comment:
it could be flaky if redis does 100k pings in less than a second
can you use sleep in lua? or just blpop? if not - use `while(true)` like you do in test_client.rs